src/opm/output/eclipse/AggregateGroupData.cpp

```diff
@@ -961,24 +967,42 @@ void assignGroupProductionTargets(const Opm::Group&        group,
     }
 }
 
+// Compatibility shim for restart output of gas-lift rates and limits.  The
+// values are intentionally discontinuous in small interval close to zero.
+template <typename SGProp>
+float getGLORate(SGProp&& sgprop, const std::optional<double>& rate)
+{
+    if (! rate.has_value()) {
+        // Defaulted rate limit (e.g., "supply" or "total").
+        return ::Opm::RestartIO::Helpers::
+            VectorItems::SGroup::Value::NoGLOLimit;
+    }
+
+    // Note: These thresholds and values are in output units.
+    const auto smallRateThreshold = 1.0e-20f;
+    const auto smallRateDefaultValue = 1.0e-6f;
+
+    const auto glo_rate =
+        sgprop(Opm::UnitSystem::measure::gas_surface_rate, rate.value());
+
+    if ((glo_rate < 0.0f) || !(glo_rate < smallRateThreshold)) {
+        // rate \not\in [0, smallRateThreshold) -> Unchanged
+        return glo_rate;
+    }
+
+    // rate \in [0, smallRateThreshold) -> smallRateDefaultValue
+    return smallRateDefaultValue;
+}
+
 template <typename SGProp, class SGrpArray>
 void assignGasLiftOptimisation(const Opm::GasLiftOpt::Group& group,
                                SGProp&&                      sgprop,
                                SGrpArray&                    sGrp)
 {
     using Ix = ::Opm::RestartIO::Helpers::VectorItems::SGroup::prod_index;
-    using M  = ::Opm::UnitSystem::measure;
 
-    sGrp[Ix::GLOMaxSupply] = sGrp[Ix::GLOMaxRate] =
-        ::Opm::RestartIO::Helpers::VectorItems::SGroup::Value::NoGLOLimit;
-
-    if (const auto& max_supply = group.max_lift_gas(); max_supply.has_value()) {
-        sGrp[Ix::GLOMaxSupply] = sgprop(M::gas_surface_rate, max_supply.value());
-    }
-
-    if (const auto& max_total = group.max_total_gas(); max_total.has_value()) {
-        sGrp[Ix::GLOMaxRate] = sgprop(M::gas_surface_rate, max_total.value());
-    }
+    sGrp[Ix::GLOMaxSupply] = getGLORate(std::forward<SGProp>(sgprop), group.max_lift_gas());
```

Comment:
I am maybe a bit overcautious here, but is it not a bit dangerous in general to std::forward the same argument twice? Maybe the receiving function should take the argument by const reference, to be sure sgprop remains unchanged from one call to the next?